aten/src/ATen/native/UpSample.h

```diff
@@ -60,10 +60,12 @@ static inline void upsample_1d_shape_check(
       ")");
 
   if (input.defined()) {
+    // Allow for empty batch size but not other dimensions
     TORCH_CHECK(
-        input.numel() != 0 && input.dim() == 3,
-        "Non-empty 3D data tensor expected but got a tensor with sizes ",
-        input.sizes());
+                (input.numel() != 0 ||
```

Comment:
The test on numel is totally unnecessary now... because if a[1] and a[2] are nonzero, numel can only be zero when a[0] is zero ;)